component/src/main/java/io/siddhi/extension/io/file/FileHandler.java

```diff
@@ -0,0 +1,311 @@
+    private SourceEventListener sourceEventListener;
+    private long monitoringInterval = 100;
+    private String listeningDirUri;
+    private FileAlterationMonitor monitor;
+    private Map<String, Long> fileObjectMap = new ConcurrentHashMap<>();
+    private static final String CURRENT_MAP_KEY = "current.map.key";
+    private List<String> fileObjectList;
+
+    @Override
+    protected ServiceDeploymentInfo exposeServiceDeploymentInfo() {
+        return null;
+    }
+
+    @Override
+    public StateFactory<FileHandlerState> init(SourceEventListener sourceEventListener,
+                                               OptionHolder optionHolder,
+                                               String[] requiredProperties,
+                                               ConfigReader configReader,
+                                               SiddhiAppContext siddhiAppContext) throws SiddhiAppValidationException {
+        this.sourceEventListener = sourceEventListener;
+        if (optionHolder.isOptionExists(Constants.URI)) {
+            listeningDirUri = optionHolder.validateAndGetStaticValue(Constants.URI);
+        }
+        //Validation for URI
+        if (listeningDirUri == null || listeningDirUri.isEmpty()) {
+            throw new SiddhiAppCreationException("URI must be provided.");
+        }
+        FileObject listeningFileObject = Utils.getFileObject(listeningDirUri);
+        try {
+            if (!listeningFileObject.exists()) {
+                throw new SiddhiAppCreationException("Directory " + listeningFileObject.getPublicURIString()
+                        + " is not found.");
+            }
+            if (listeningFileObject.isFile()) {
+                throw new SiddhiAppCreationException("URI must belongs to a folder");
+            }
+            listeningDirUri = listeningFileObject.getName().getPath();
+        } catch (FileSystemException e) {
+            throw new SiddhiAppValidationException("Directory " + listeningFileObject.getPublicURIString()
+                    + " is not found.", e);
+        }
+
+        // Validation for fileNameList
+        String fileNameList = optionHolder.validateAndGetStaticValue(Constants.FILE_NAME_LIST, EMPTY_STRING);
+        fileNameList = fileNameList.replaceAll("\\s", "");
+        fileObjectList = Arrays.asList(fileNameList.split(","));
+        for (int i = 0; i < fileObjectList.size(); i++) {
+            String fileObjectPath = listeningDirUri + "/" + fileObjectList.get(i);
+            listeningFileObject = Utils.getFileObject(fileObjectPath);
+            try {
+                if (!listeningFileObject.exists()) {
+                    throw new SiddhiAppCreationException("File/Folder " +
+                            listeningFileObject.getPublicURIString() + " is not found.");
+                }
+            } catch (FileSystemException e) {
+                log.error("File/Folder " + listeningFileObject.getPublicURIString() + " is not found");
```

Comment:
@Ajanthy the exception object to the log.error() function as well. eg: log.error("abcd", e)

Reply:
fixed in 5f2348a3cec4571d14c1f3f939dfddcbadd466e8